### PR TITLE
Apim 4343 filter subscription

### PR DESCRIPTION
**Pull request by jansobczynski**
## Issue

https://gravitee.atlassian.net/browse/APIM-4343

## Description

Add status filter subscription feature

## Additional context
Sceenshot:
Screenshot 2024-05-27 at 11 14 02 AM

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-jaqnxnsqbe.chromatic.com)

### CODE REVIEW

File: gravitee-apim-portal-webui-next/src/entities/subscription/subscription.ts [hunk 1]
```diff
@@ -35,6 +35,11 @@ export interface SubscriptionData {
 export interface SubscriptionLinks {
   self?: string;
 }
+
 export interface SubscriptionMetadata {
   [key: string]: { [key: string]: object | string };
 }
+
+export const Subscriptions = {
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/entities/subscription/subscription.ts]: The subscription statuses will always match these values: https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/resources/portal-openapi.yaml#L4366

The objects exported from this file should match what we expect to receive from the back. It is then up to the renderer/`api-tab-subscriptions` component to format the information (aka turn it into capitalize or lowercase).

File: gravitee-apim-portal-webui-next/src/app/api-details/api-tab-subscriptions/api-tab-subscriptions.component.ts [hunk 2]
```diff
@@ -14,35 +14,39 @@
...
+  public displayedColumns: string[] = ['application', 'plan', 'status', 'expand'];
+  public subscriptionStatusesList = Subscriptions.statuses;
+  public subscriptionsStatus = new FormControl();
+  public subscriptionsList$!: Observable<SubscriptionData[]>;
 
-  constructor(private capitalizeFirstPipe: CapitalizeFirstPipe) {}
+  private subscriptionService = inject(SubscriptionService);
 
-  ngOnInit(): void {
+  constructor(private capitalizeFirstPipe: CapitalizeFirstPipe) {
```
Comment (jourdiw) [inline gravitee-apim-portal-webui-next/src/app/api-details/api-tab-subscriptions/api-tab-subscriptions.component.ts]: What was your reasoning to have the service injected above the constructor and the pipe injected within the constructor?

Comment (jansobczynski) [reply gravitee-apim-portal-webui-next/src/app/api-details/api-tab-subscriptions/api-tab-subscriptions.component.ts]: Fixed, I moved one into constructor as well

File: gravitee-apim-portal-webui-next/src/entities/subscription/subscription.ts (hunk 1 above)
Comment (jansobczynski) [reply gravitee-apim-portal-webui-next/src/entities/subscription/subscription.ts]: Done

**Review by jourdiw — APPROVED** [review]

**Pull request merged**